src/responder/pam/pamsrv_passkey.h

```diff
@@ -36,14 +36,19 @@ enum passkey_user_verification {
     PAM_PASSKEY_VERIFICATION_INVALID
 };
 
-errno_t passkey_local(TALLOC_CTX *mem_ctx,
-                             struct tevent_context *ev,
-                             struct pam_ctx *pam_ctx,
-                             struct pam_auth_req *preq,
-                             struct pam_data *pd);
-errno_t passkey_kerberos(struct pam_ctx *pctx,
-                         struct pam_data *pd,
-                         struct pam_auth_req *preq);
+/* Operations when calling passkey child */
+enum passkey_child_op {
+    PAM_PASSKEY_OP_PREFLIGHT,
+    PAM_PASSKEY_OP_KERBEROS_AUTH,
+    PAM_PASSKEY_OP_LOCAL_AUTH,
+    PAM_PASSKEY_OP_INVALID
+};
+
+struct pam_preflight_data {
+    int attempts;
+    bool pin_required;
+    bool obtained;
```

Comment:
Since we are changing the structure name I'd also change this to `preflight_obtained` or `preflight_executed`